### PR TITLE
improved script, other minor improvements

### PR DESCRIPTION
**Pull request by twblackwell**
- scrapped outdated "adding script back to bill page"
- added script and bill number back to bill page
- fixed UI when going from full screen to smaller screens
- fixed session issue when refreshing and reps show incorrectly
- other UI improvements to bill page

### CODE REVIEW

File: frontend/src/components/Bill/Bill.jsx [hunk 1]
```diff
@@ -1,186 +1,164 @@
...
-    1 +
-    items[activeVersion].coSponsors.size +
-    items[activeVersion].multiSponsors.size
-  );
-}
+// Helper functions
+const getSponsorNumber = (billData) => {
+  const { activeVersion, amendments: { items } } = billData;
+  return 1 + items[activeVersion].coSponsors.size + items[activeVersion].multiSponsors.size;
+};
```
Comment (elliott-king) [inline frontend/src/components/Bill/Bill.jsx:13]: This is not an actual change, just a formatting change. Can you go into your setting and find the value for this (check both the user & workspace settings):

Screenshot 2024-10-02 at 1 48 00 PM

Also, the `defaultFormatter` for `javascriptreact`?

**Review by elliott-king — CHANGES_REQUESTED** [review]

File: frontend/src/components/Bill/Bill.jsx (hunk 1 above)
Comment (elliott-king) [reply frontend/src/components/Bill/Bill.jsx:13]: There's so much formatting changes here that it's hard to parse what the actual changes are.

Comment (twblackwell) [reply frontend/src/components/Bill/Bill.jsx:12]: I removed some of the unnecessary formatting changes. Also, my print width is 80, is that okay?

File: frontend/src/components/Bill/LocForm.jsx [hunk 2]
```diff
@@ -19,20 +19,37 @@ const LocForm = ({ sponsorNames }) => {
...
     }
+
     const sessionLongitude = window.sessionStorage.getItem('longitude');
     if (sessionLongitude) {
       setLongitude(JSON.parse(sessionLongitude));
     }
-  }, []);
+  }, [sponsorNames]);
 
   const reset = () => {
```
Comment (elliott-king) [inline frontend/src/components/Bill/LocForm.jsx:54]: Clicking "change location" throws an error: `storedRepresentatives.filter is not a function`, which is happening b/c the below is setting it to `{}` instead of `[]`.

I'd recommend simply deleting all of the items in sessionStorage instead of setting them to nullish values.

**Review by elliott-king — CHANGES_REQUESTED** [review]

**Comment by elliott-king** [comment]
I also have recommendations, but these are OOS. You can implement them now or we can make a project ticket for them, and one of us can pick them up later:
- We should use `localStorage` instead of `sessionStorage`. This logic doesn't persist in a new tab currently.
- We should use clearer  storage variable names. eg, "sunrise-placeName", "sunrise-latitude"

**Comment by agreffard** [comment]
> I also have recommendations, but these are OOS. You can implement them now or we can make a project ticket for them, and one of us can pick them up later:
> 
> * We should use `localStorage` instead of `sessionStorage`. This logic doesn't persist in a new tab currently.
> * We should use clearer  storage variable names. eg, "sunrise-placeName", "sunrise-latitude"

Legally we are not allowed to persistently keep the data coming from the API, that's why I used the sessionStorage and not localStorage
https://docs.mapbox.com/api/search/geocoding/#storing-geocoding-results

I agree for the variable naming

**Comment by elliott-king** [comment]
> Legally we are not allowed to persistently keep the data coming from the API, that's why I used the sessionStorage and not localStorage https://docs.mapbox.com/api/search/geocoding/#storing-geocoding-results

localStorage is not permanent. It erases on window/browser close. Additionally, we are not setting the optional `permanent` parameter mentioned here. Their language is really confusing - see also [here](https://www.mapbox.com/geocoding) and [here](https://docs.mapbox.com/help/faq/what-is-the-difference-between-temporary-and-permanent-geocoding/), but I'm 99% sure we're fine.

**Comment by agreffard** [comment]
I think localStorage is more persistent than that. It will only clear on close if you're on private mode but otherwise will stay persistent on window close, restart or even after clearing cookies.
That's why I was not sure  and found sessionStorage to be a good compromise (technically we are indeed "storing" the results).
But I agree the language is very confusing.
I'm ok with localStorage though if you all understand that the restrictions do not apply to us

**Comment by elliott-king** [comment]
> I think localStorage is more persistent than that. [...]

That's correct, my bad. I think the "permanent" they use is referring to something server-side (like redis, sql). I checked their Discord, they have a similar question, but it has never been answered  ¯\\_(ツ)_/¯ You can blame me if the cops ever come for you.

**Review by twblackwell — COMMENTED** [review]
I think we're set now?

**Review by elliott-king — APPROVED** [review]
